frame/node-authorization/src/lib.rs

```diff
@@ -442,24 +510,35 @@ pub mod pallet {
 }
 
 impl<T: Config> Pallet<T> {
-	fn initialize_nodes(nodes: &Vec<(PeerId, T::AccountId)>) {
-		let peer_ids = nodes.iter().map(|item| item.0.clone()).collect::<BTreeSet<PeerId>>();
-		WellKnownNodes::<T>::put(&peer_ids);
+	fn initialize_nodes(nodes: &Vec<(PeerId, T::AccountId)>) -> Result<(), Error<T>> {
+		let mut peer_ids = BoundedBTreeSet::new();
+		let mut owners = Vec::new();
+		for (peer_id, account_id) in nodes {
+			let peer_id = BoundedPeerId::<T>::try_from(peer_id.clone())?;
+			peer_ids.try_insert(peer_id.clone()).map_err(|_| Error::<T>::TooManyNodes)?;
+			owners.push((peer_id, account_id));
```

Comment:
```suggestion
			Owners::<T>::insert(peer_id, account_id);
```
and remove the `owners` vector?